### DISCUSSION TITLE
Error: About downloading Waffle

### DISCUSSION THREAD

**Discussion by TimeKeyRoll** [NomicFoundation/hardhat#3040] [Troubleshooting] [answered]
Hi! since we have this new feature of hardhat which is `hardhat-toolbox` and which comes with 
```
HARDHAT- TOOLBOX

Ethers.js and harhdat-ethers
chai and mocha
hardhat chai matchers
hardhat network helpers
harahd etherscan
hardhat-gas-reporter
solidity-coverage
typechain
```
 so does that we dont have to download them seprately if we have tool box installed?  

if its right them im trying to download waffle but it does not getting download. does waffle also comes with `hardhat-toolbox`?

kindy help

**Comment by fvictorio** [comment, marked as answer]
> so does that we dont have to download them seprately if we have tool box installed?

That depends. If you are using npm 7 or later, they will be automatically installed when you install the toolbox. If not, you have to install them (similar to how you have to install ethers if you want to use the hardhat-ethers plugin). Check the instructions [here](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-toolbox).

> does waffle also comes with hardhat-toolbox

No, Waffle is not included in the Toolbox. It was replaced by [our own chai matchers](https://hardhat.org/hardhat-chai-matchers/docs/overview).
